pkg/hubble/parser/sock/parser.go

```diff
@@ -0,0 +1,221 @@
+	decoded.DestinationNames = p.resolveNames(srcEndpoint.GetID(), dstIP)
+	decoded.Type = flowpb.FlowType_SOCK
+	decoded.EventType = decodeCiliumEventType(sock.Type, sock.XlatePoint)
+	decoded.SockXlatePoint = flowpb.SocketTranslationPoint(sock.XlatePoint)
+	decoded.SocketCookie = sock.SockCookie
+	decoded.CgroupId = sock.CgroupId
+	decoded.Summary = decodeSummary(sock)
+	return nil
+}
+
+func decodeIPVersion(flags uint8) flowpb.IPVersion {
+	if (flags & monitor.TraceSockNotifyFlagIPv6) != 0 {
+		return flowpb.IPVersion_IPv6
+	}
+	return flowpb.IPVersion_IPv4
+}
+
+func (p *Parser) decodeSrcIP(cgroupId uint64, ipVersion flowpb.IPVersion) net.IP {
+	if p.cgroupGetter != nil {
+		if m := p.cgroupGetter.GetPodMetadataForContainer(cgroupId); m != nil {
+			for _, podIP := range m.IPs {
+				isIPv6 := strings.Contains(podIP, ":")
+				if isIPv6 && ipVersion == flowpb.IPVersion_IPv6 ||
+					!isIPv6 && ipVersion == flowpb.IPVersion_IPv4 {
+					return net.ParseIP(podIP)
+				}
+			}
+			p.log.WithField("id", cgroupId).WithField("pod", m.Namespace+"/"+m.Name).Debug("no matching IP for pod")
+		}
+	}
+
+	return nil
+}
+
+func decodeL3(srcIP, dstIP net.IP, ipVersion flowpb.IPVersion) *flowpb.IP {
+	var srcIPStr, dstIPStr string
+	if srcIP != nil {
+		srcIPStr = srcIP.String()
+	}
+	if dstIP != nil {
+		dstIPStr = dstIP.String()
+	}
+
+	return &flowpb.IP{
+		Source:      srcIPStr,
+		Destination: dstIPStr,
+		IpVersion:   ipVersion,
+	}
+}
+
+func decodeL4(proto uint8, srcPort, dstPort uint16) *flowpb.Layer4 {
+	switch proto {
+	case monitor.L4ProtocolTCP:
+		return &flowpb.Layer4{
+			Protocol: &flowpb.Layer4_TCP{
+				TCP: &flowpb.TCP{
+					SourcePort:      uint32(srcPort),
+					DestinationPort: uint32(dstPort),
+				},
+			},
+		}
+	case monitor.L4ProtocolUDP:
+		return &flowpb.Layer4{
+			Protocol: &flowpb.Layer4_UDP{
+				UDP: &flowpb.UDP{
+					SourcePort:      uint32(srcPort),
+					DestinationPort: uint32(dstPort),
+				},
+			},
+		}
+	}
+
+	return nil
+}
+
+func (p *Parser) resolveNames(epID uint32, ip net.IP) (names []string) {
+	if p.dnsGetter != nil {
+		return p.dnsGetter.GetNamesOf(epID, ip)
+	}
+
+	return nil
+}
+
+func (p *Parser) decodeService(ip net.IP, port uint16) *flowpb.Service {
+	if p.serviceGetter != nil {
+		return p.serviceGetter.GetServiceByAddr(ip, port)
+	}
+
+	return nil
+}
+
+func decodeRevNat(xlatePoint uint8) *wrapperspb.BoolValue {
+	switch xlatePoint {
+	case monitor.XlatePointPreDirectionFwd,
+		monitor.XlatePointPostDirectionFwd:
+		return &wrapperspb.BoolValue{Value: false}
+	case monitor.XlatePointPreDirectionRev,
+		monitor.XlatePointPostDirectionRev:
+		return &wrapperspb.BoolValue{Value: true}
+	}
+
+	return nil
+}
+
+func decodeCiliumEventType(eventType, subtype uint8) *flowpb.CiliumEventType {
+	return &flowpb.CiliumEventType{
+		Type:    int32(eventType),
+		SubType: int32(subtype),
+	}
+}
+
+func decodeSummary(sock *monitor.TraceSockNotify) string {
```

Comment:
Yes. The summary refers to the string in parentheses in the following screenshot:

![image](https://user-images.githubusercontent.com/50564/197740190-78831bd9-5fba-4d0f-9cda-0fa24c5ac884.png)

Since we don't have more information on the actual contents of the packet for trace sock events (e.g. TCP flags), just displaying the protocol seems to be the most consistent with the existing output. But I'm open to suggestions.

Reply:
`Summary` is marked as deprecated, and `decodeSummary` only provides info that we already know through `L4`. IMHO this logic belongs to the client, but I don't feel too strongly about it.